test/integration/controller/jobs/job/job_controller_test.go

```diff
@@ -1832,6 +1895,150 @@ var _ = ginkgo.Describe("Job controller interacting with scheduler", ginkgo.Orde
 	})
 })
 
+var _ = ginkgo.Describe("Job controller interacting with scheduler when waitForPodsReady enabled", ginkgo.Ordered, ginkgo.ContinueOnFailure, func() {
```

Comment:
I fear that there is high potential for flakiness here.

The ideal solution is to manipulate the passing of time in the test, if you can pass a fake time to every controller.

OTOH, we are not too concerned with what the scheduler is doing, so we could potentially remove it from the equation. What we are concerned mostly is the interaction between the job controller and the workload reconciler.

@mimowo do you have any suggestions on how to improve this test?

Reply:
With current solution I think we can't manipulate the passing of time in the test. Inside worload_controller we are using another incapsulated waitForPodsReadyConfig.

Reply:
With scheduler I think it's more realistic case.